app/src/main/java/me/ccrama/redditslide/Activities/MediaView.java

```diff
@@ -1226,4 +1229,23 @@ public void onFolderSelection(@NonNull FolderChooserDialogCreate dialog,
     @Override
     public void onFolderChooserDismissed(@NonNull FolderChooserDialogCreate dialog) {
     }
+
+
+    //this function raises a toast in case the size of the requested file is greater than the alert limit set
+    //the idea is to use the text view that we already have in the layout which stores the size of the media file.
+    //we compare the set alert limit with the text in the textview and process accordingly.
+    public void saveMedia(View view) {
+        View relLayout = findViewById(R.id.gifheader);
+        ImageView save = (ImageView)relLayout.findViewById(R.id.save);
+        TextView txtMediaSize = (TextView)relLayout.findViewById(R.id.size);
+        if (txtMediaSize.getText()!=null && txtMediaSize.getText().toString().length()!=0){
+            String size = txtMediaSize.getText().toString();
+            size=size.substring(0,size.length()-1);  // to remove 'MB'
+            SharedPreferences preferences = getSharedPreferences("prefs", Context.MODE_PRIVATE);
+            String alertLimit = preferences.getString("limit","1000");  //we can set 1000 MB to be default data alert limit
+            if(Integer.parseInt(size)>Integer.parseInt((alertLimit))){
+                Toast.makeText(MediaView.this,R.string.err_media_greaterThanLimit,Toast.LENGTH_SHORT).show();
+            }
+        }
+    }
```

Comment:
Also this doesn't at all close #3074. The issue creator's request was to use a HEAD request to show the size of the media on the post cards. This implementation only deals with when the save button is clicked, which is effectively redundant as the post is still being loaded regardless (because it's a separate task). The actual post loading is what concerns the issue creator, not saving the media.

Reply:
Okayy, I'll work on this and make another PR using the mentioned approach in sometime :)